src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -303,7 +303,7 @@ Definitions.CHECKSTYLE_BUNDLE, getClass(),
     private Map<String, Set<String>> generateThirdPartyNameToFullModuleName(ClassLoader loader) {
         Map<String, Set<String>> returnValue;
         try {
-            returnValue = ModuleReflectionUtil.getCheckstyleModules(packages, loader).stream()
+            returnValue = ModuleReflectionUtil.getValidCheckstyleClasses(packages, loader).stream()
```

Comment:
This is needed so we can keep the cache of `thirdPartyNameToFullModuleNames`. We need someone to pull in classes to put into the cache, and removing it completely will just load all classes in the classpath. `getCheckstyleModules` was always the method that was making this restrictive.

Passing a Predicate like originally suggested will not assist, one because of the cache. The cache would need the predicate to know what should be populated. However, each walker needs to pass its own predicate which would complicate caching.
Another reason a Predicate will make things more complex is we are currently throwing exceptions if we find multiple classes with the same name. So we would have to add more logic and loops to determine if a Predicate would help any, while breaking some compatibility to allow the Predicate to be given to this class.
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java#L280

Checker and TreeWalker is already ensuring it is only dealing with the children it wants, so this is why more code isn't needed.
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L483
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java#L131

Reply:
Solution is approved, please finalize code changes.
Makes sense to create issue with description of this comment.

Do we need isCheckstyleModule method?

Reply:
`isCheckstyleModule` is still needed for other utils and satisfies the check preventing methods from being too big.

Reply:
I see only single usage (not in tests):
![image](https://user-images.githubusercontent.com/812984/198294469-57e5d5e1-6f3d-4cc1-b0ec-b2413dff2ba7.png)

please double check your side.
I think we should remove this method.

Reply:
I pushed commit with removal of this method.
I still do not understand why we need two different methods to load modules.

So now, can we rename getValidCheckstyleClasses to getCheckstyleModules ?

Reply:
My main reason against removal is because `CheckUtil` is our go-to place to get all actual checkstyle modules in the repo to use for documentation and other "what are our checks area".

We are changing this method from `getCheckstyleModules` to `getValidCheckstyleClasses`. So for a trimmed example, instead of getting all filesets we are now changing it to get everything that looks like a module, but we have not confirmed ourselves if it is. So we are possibly getting more classes that are not our modules. We would have to change this method some to ensure we are only reporting our modules in 

That said, these test method is trying to become more open to others repos like sevntu to be used there and maybe it could be used by checkstyle-extras for its own modules. I have an open issue on this, but sevntu is stuck because of eclipse-cs. I am surprised that the test change did not cause any failures.

I want to look at more in my local, but i think I will change my mind and make the change.

Reply:
>but we have not confirmed ourselves if it is

If we are not confident, we can not merge this PR.

Reply:
I confirmed locally, it seems the method returns the same list for our repo. So I pushed the update. Please look over and I will ensure the CI passes.